pkg/services/ldap/ldap.go

```diff
@@ -192,6 +202,33 @@ func (auth *Auth) SyncUser(query *models.LoginUserQuery) error {
 	return nil
 }
 
+// DisableExternalUser marks external user as disabled in Grafana db
+func (auth *Auth) DisableExternalUser(username string) error {
+	// Check if external user exist in Grafana
+	userQuery := &models.GetExternalUserInfoByLoginQuery{
+		LoginOrEmail: username,
+	}
+
+	if err := bus.Dispatch(userQuery); err != nil {
+		return err
+	}
+
+	userInfo := userQuery.Result
+	if userInfo.AuthModule == "ldap" && !userInfo.IsDisabled {
```

Comment:
It seems logical, but just want to emphasize that - disabled users could only be ldap ones. Is there any cases we foresee in the future that we might use for other cases?

Reply:
Can we use early return technic here?

Reply:
We want to make it possible to disable non-external users from the Admin UI. This extra check (`userInfo.AuthModule == "ldap"`) is required because `GetExternalUserInfoByLoginQuery` returns not only LDAP users. So potentially, if you're using OAuth, for instance, alongside with LDAP, and existing OAuth user is trying to login, it will be disabled. I'm not 100% sure, but from my point of view, it looks like this.

Reply:
Outside of the discussion above: we should also create a const for the ldap string.

Reply:
@xlson I added 
```go
const AuthModuleLDAP = "ldap"
```
to the models. But should I add other AuthModule-related constants like `authproxy`, `generic_oauth`, maybe empty string as well?

Reply:
I would only add those that we actively work with and make another PR to clean up the others, or just wait until we work with the others.